File: backend/app/schemas/audit.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class AuditLogItemRead(BaseModel):
    id: int
    created_at: datetime
    user_id: Optional[int]
    user_name: Optional[str]
    entity_type: str
    entity_id: str
    action_type: str
    old_value: Optional[dict]
    new_value: Optional[dict]


class AuditLogListResponse(BaseModel):
    items: list[AuditLogItemRead]
    total: int
    limit: int
    offset: int
    action_types: list[str]
    entity_types: list[str]

File: frontend/src/contracts/api/audit.ts
export type AuditLogItem = {
  id: number;
  created_at: string;
  user_id: number | null;
  user_name: string | null;
  entity_type: string;
  entity_id: string;
  action_type: string;
  old_value: Record<string, unknown> | null;
  new_value: Record<string, unknown> | null;
};

export type AuditLogResponse = {
  items: AuditLogItem[];
  total: number;
  limit: number;
  offset: number;
  action_types: string[];
  entity_types: string[];
};
